samcli/commands/local/invoke/cli.py

```diff
@@ -98,6 +99,8 @@ def do_cli(ctx, function_identifier, template, event, no_event, env_vars, debug_
         raise UserException("Function {} not found in template".format(function_identifier))
     except (InvalidSamDocumentException, OverridesNotWellDefinedError) as ex:
         raise UserException(str(ex))
+    except DockerImagePullFailedException as ex:
```

Comment:
I elected for the stack trace for these. Mainly because I didn't have a good way to surface this. Take `start-api` for example. When calling a localhost endpoint, the response from the API will be a service error. We could catch the error and 'pretty print' something but figured the stack trace is enough for now (since this is the current behavior). 

I would like to think more through this case in general (how we behave offline) but not halt that on providing some benefit now. I have no problem adding this now, just explaining my thought process of why I initially didn't.

Reply:
I see what you mean. If you don't catch the exception, then it is going to crash the process and print a stack trace right? 

It would be rather better to print a proper error message and terminate the CLI than a crash.

Reply:
The process doesn't crash. We rely on Flask for this. Flask will see the exception and then wrap this in a service 500 error: https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/local/lambda_service/local_lambda_invoke_service.py#L114. 

This is the same behavior for `start-api`. We only catch the errors we care about and handle them specifically, everything else gets rolled up into a 500 Error. The output will still show the stack trace of the error in the terminal.

Reply:
Ah yes, I forgot about Flask. This makes sense and is consistent with a service-like behavior.